### PR TITLE
[rayci]  add option to mount bk agent when enabled.

### PR DESCRIPTION
**Pull request by aslonnie**
so that in other previledged, non oss modes, we can use
buildkite-agent from the system

### CODE REVIEW

File: raycicmd/config.go [hunk 1]
```diff
@@ -51,6 +55,9 @@ type config struct {
 
 	// SkipTags is the list of tags that will always be skipped.
 	SkipTags []string `yaml:"skip_tags"`
+
+	// Additional docker plugin configs.
+	DockerPlugin *dockerPluginConfig `yaml:"docker_plugin"`
```
Comment (can-anyscale) [inline raycicmd/config.go:60]: It's a bit unclear which field in this list is required vs optional; maybe put (Optional) for this field in the comment.

Also I get the impression that ForgeDirs is not used when I create the config for release test pipeline.

**Review by can-anyscale — APPROVED** [review]

File: raycicmd/config.go (hunk 1 above)
Comment (aslonnie) [reply raycicmd/config.go:60]: yeah. most fields are optional, only some fields are mandatory.

forge dir is deprecated now after we added the wanda rules. Let me file a PR to remove that.

Comment (aslonnie) [reply raycicmd/config.go:115]: added many comments. will remove the forgedir in a follow up.

**Pull request merged**